VestingInstance.sol

```diff
@@ -0,0 +1,273 @@
+    uint256 public _vestingPeriodSec;
+
+    // timestamps for start and cliff starting points. Timestamps are expressed in UNIX time, the same units as block.timestamp.
+    uint256 public _cliffStartTime;
+    uint256 public _vestingStartTime;
+
+    // indicates if the contract is revokable
+    bool public _revocable;
+
+    // revoking address and refund destination
+    address public _refundDestination;
+    address public _revoker;
+
+    // indicates how much of the vested amount has been released for withdrawal (i.e. withdrawn)
+    uint256 public _currentlyReleased;
+
+    // indicates if the vesting has been revoked. false by default
+    bool public _revoked;
+
+    // the time at which the revocation has taken place
+    uint256 public _revokeTime;
+
+    /**
+     * @notice A constructor for initialising a new instance of a Vesting Schedule contract
+     * @param beneficiary address of the beneficiary to whom vested tokens are transferred
+     * @param vestingAmount the amount that is to be vested by the contract
+     * @param vestingCliff duration in seconds of the cliff in which tokens will begin to vest
+     * @param vestingStartTime the time (as Unix time) at which point vesting starts
+     * @param vestingPeriodSec duration in seconds of the period in which the tokens will vest
+     * @param vestAmountPerPeriod the vesting amound per period where period is the vestingAmount distributed over the vestingPeriodSec
+     * @param revocable whether the vesting is revocable or not
+     * @param revoker address of the person revoking the vesting
+     * @param refundDestination address of the refund receiver after the vesting is deemed revoked
+     * @param vestingContractOwner the owner that of the vesting contract
+     */
+    constructor (address beneficiary,
+                uint256 vestingAmount,
+                uint256 vestingCliff,
+                uint256 vestingStartTime,
+                uint256 vestingPeriodSec,
+                uint256 vestAmountPerPeriod,
+                bool    revocable,
+                address revoker,
+                address refundDestination,               
+                address vestingContractOwner) public {
+
+        // do some basic checks
+        require(vestingAmount > 0, "Amount must be positive");
+        require(beneficiary != address(0), "Beneficiary is the zero address");
+        require(refundDestination != address(0), "Refund destination is the zero address");
+        // solhint-disable-next-line max-line-length
+        require(vestingCliff <= vestingPeriodSec, "Vesting cliff is longer than duration");
+        // solhint-disable-next-line max-line-length
+        require(vestingPeriodSec > 0, "Vesting period is 0 s.");
+        // solhint-disable-next-line max-line-length
+        require(vestAmountPerPeriod <= vestingAmount, "Vesting amount per period is greater than the total vesting amount");
+        // solhint-disable-next-line max-line-length
```

Comment:
I'd remove the solhint disables for the celo repo submission

Reply:
lines removed